integration_tests/juniper_tests/src/codegen/impl_scalar.rs

```diff
@@ -0,0 +1,42 @@
+use fnv::FnvHashMap;
+#[cfg(test)]
+use juniper::{
+    self, parser::ScalarToken, DefaultScalarValue, GraphQLType, InputValue, ParseScalarResult,
+    ParseScalarValue, Value,
+};
+
+struct UserId(String);
+
+// TODO Trait that the macro handles, move to proper location! Naming?
```

Comment:
I don't think we actually need a trait..we can just generate the implementation directly (the current macro doesn't use a trait, right?). For example, there is no `GraphQLObject` trait and instead it just implements `GraphQLType` if I recall correctly.